### PR TITLE
lightning: standardize error codes and messages

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32239

Problem Summary:
 
Standardize error codes and messages for Lightning.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/common/errors.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	ErrEncodeKV           = errors.Normalize("encode kv error in file %s at offset %d", errors.RFCCodeText("Lightning:Restore:ErrEncodeKV"))
+	ErrAllocTableRowIDs   = errors.Normalize("allocate table row id error", errors.RFCCodeText("Lightning:Restore:ErrAllocTableRowIDs"))
+	ErrInvalidMetaStatus  = errors.Normalize("invalid meta status: '%s'", errors.RFCCodeText("Lightning:Restore:ErrInvalidMetaStatus"))
+	ErrTableIsChecksuming = errors.Normalize("table '%s' is checksuming", errors.RFCCodeText("Lightning:Restore:ErrTableIsChecksuming"))
+)
+
+type withStack struct {
+	error
+	errors.StackTracer
+}
```
Comment (kennytm) [inline br/pkg/lightning/common/errors.go:95]: why do we have to implement these 🤔

Comment (gozssky) [reply br/pkg/lightning/common/errors.go:95]: To keep the original stack trace when convert BR code to Lightning

**Review by glorv — COMMENTED** [review]
rest LGTM

File: errors.toml [hunk 2]
```diff
@@ -226,6 +226,271 @@ error = '''
...
+'''
+
+["Lightning:Checkpoint:ErrUpdateCheckpoint"]
+error = '''
+update checkpoint error
+'''
+
+["Lightning:Common:ErrInvalidArgument"]
+error = '''
+invalid argument
```
[the remark this answers is not in the document]
Comment (gozssky) [reply errors.toml:276]: The error is from `flag.Parse`. It's not it's convenient to parse which field is invalid. But the error message itself is detailed.

**Review by glorv — APPROVED** [review]
LGTM

**Comment by glorv** [comment]
/cc @sunzhaoyang 
PTAL

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**